### PR TITLE
new global option: USE_64_BIT_INDICES

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

**Comment by koecher** [comment]
looks nice. Are there more packages like slepc or p4est which needs this information as well? (A question to the experts). 

Otherwise, just merge.

**Review by koecher — CHANGES_REQUESTED** [review]

**Comment by koecher** [comment]
@tjhei, in this setting, I believe that slepc is using the best fit "confopts" from petsc, but this is not the way we should go for candi...

File: deal.II-toolchain/packages/slepc.package [hunk 1]
```diff
@@ -40,11 +40,13 @@ package_specific_setup () {
     # make sure no other invalid SLEPC_DIR is set:
     unset SLEPC_DIR
 
-    ${PYTHON_INTERPRETER} ./configure --prefix=${INSTALL_PATH} ${CONFOPTS}
+    ${PYTHON_INTERPRETER} ./configure --prefix=${INSTALL_PATH}
```
[the remark this answers is not in the document]
Comment (tjhei) [reply deal.II-toolchain/packages/slepc.package:43]: No, CONFOPTS is empty and you do not (can not) specify Petsc options here. Slepc grabs them from Petsc automatically. I don't think there is a different way to do this.

[the remark this answers is not in the document]
Comment (koecher) [reply deal.II-toolchain/packages/slepc.package:43]: ah, snap - those were different files - sorry. Maybe we haven't reset CONFOPTS correctly, or there is something in the defaults which now breaks slepc, if this line is the deal-breaker.

[the remark this answers is not in the document]
Comment (koecher) [reply deal.II-toolchain/packages/slepc.package:43]: anyhow, we should make it possible, that CONFOPTS is still pushed to the configure of slepc as well

**Comment by tjhei** [comment]
I think you misunderstood: I deleted ``${CONFOPTS}`` because it is an unused/empty variable. SLEPc does not need PETSc arguments but grabs them from the PETSc installation. It is in fact an error to provide PETSc options here.

CONFOPTS is reset in candi.sh here: https://github.com/dealii/candi/blob/609ef27b96f92dc402ce59b1d4509f23a676cc89/candi.sh#L1107

**Comment by tjhei** [comment]
I am fairly confident that this is correct, so I am merging.

**Pull request merged**